tools/hsmtool.c

```diff
@@ -469,7 +490,9 @@ int main(int argc, char *argv[])
 		if (argc < 3)
 			show_usage(argv[0]);
 
-		return dumponchaindescriptors(argv[2], argc > 3 ? argv[3] : NULL);
+		return dumponchaindescriptors(argv[2],
+					      argc > 3 && !streq(argv[3], "") ? argv[3] : NULL,
+					      argc > 4 && streq(argv[4], "private"));
```

Comment:
it's weird to me that you have to physically type in the phrase "private" and not look for a true/false here.

Reply:
I find it better: at usage typing `testnet` is way more meaningful than `true`

Reply:
Update the parameter name then. As written, the parameter `testnet` suggests to say 'flag true if testnet'. 

Same thing for `private`.